### PR TITLE
Add documentation for Druids catalog extension

### PR DESCRIPTION
**Pull request by zachjsh**
This pr adds docs for Druids catalog extension

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: docs/development/extensions-core/catalog.md [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+```json
+{
+    "version": 12345687
+}
+```
+
+</TabItem>
+<TabItem value="2" label="400 BAD REQUEST">
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:115]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+##### URL
+
+`GET` `/druid/coordinator/v1/catalog/schemas/{schema}/tables/{name}
+
+##### Responses
+
+<Tabs>
+
+<TabItem value="1" label="200 SUCCESS">
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:199]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+
+*Successfully retrieved corresponding table's [TableSpec](#tablespec)*
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:203]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+</TabItem>
+<TabItem value="2" label="400 BAD REQUEST">
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:207]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+```json
+{
+    "error": "A well-defined error code.",
+    "errorMessage": "A message with additional details about the error."
+}
+```
+</TabItem>
+<TabItem value="3" label="500 INTERNAL SERVER ERROR">
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:219]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 6]
```diff
@@ -0,0 +1,471 @@
...
+##### Sample request
+
+The following example shows how to retrieve a table name `test_table` in schema `druid`
+
+```shell
+curl "http://ROUTER_IP:ROUTER_PORT/druid/coordinator/v1/catalog/schemas/druid/tables/test_table"
+```
+
+##### Sample response
+<details>
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:241]:
```suggestion

<details>
```

File: docs/development/extensions-core/catalog.md [hunk 7]
```diff
@@ -0,0 +1,471 @@
...
+
+`DELETE` `/druid/coordinator/v1/catalog/schemas/{schema}/tables/{name}
+
+##### Responses
+
+<Tabs>
+
+<TabItem value="1" label="200 SUCCESS">
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:300]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 8]
```diff
@@ -0,0 +1,471 @@
...
+*No response body*
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:303]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 9]
```diff
@@ -0,0 +1,471 @@
...
+</TabItem>
+<TabItem value="2" label="400 BAD REQUEST">
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:307]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 10]
```diff
@@ -0,0 +1,471 @@
...
+```json
+{
+    "error": "A well-defined error code.",
+    "errorMessage": "A message with additional details about the error."
+}
+```
+</TabItem>
+<TabItem value="3" label="500 INTERNAL SERVER ERROR">
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:319]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 11]
```diff
@@ -0,0 +1,471 @@
...
+```shell
+curl -X 'DELETE' "http://ROUTER_IP:ROUTER_PORT/druid/coordinator/v1/catalog/schemas/druid/tables/test_table"
+```
+
+##### Sample response
+
+No response body
+
+#### retrieve list of schema names
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:344]:
```suggestion
#### Retrieve list of schema names
```

File: docs/development/extensions-core/catalog.md [hunk 12]
```diff
@@ -0,0 +1,471 @@
...
+
+`GET` `/druid/coordinator/v1/catalog/schemas
+
+##### Responses
+
+<Tabs>
+
+<TabItem value="1" label="200 SUCCESS">
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:358]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 13]
```diff
@@ -0,0 +1,471 @@
...
+*Successfully retrieved list of schema names*
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:361]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 14]
```diff
@@ -0,0 +1,471 @@
...
+</TabItem>
+<TabItem value="2" label="400 BAD REQUEST">
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:364]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 15]
```diff
@@ -0,0 +1,471 @@
...
+```json
+{
+    "error": "A well-defined error code.",
+    "errorMessage": "A message with additional details about the error."
+}
+```
+</TabItem>
+<TabItem value="3" label="500 INTERNAL SERVER ERROR">
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:377]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 16]
```diff
@@ -0,0 +1,471 @@
...
+  "INFORMATION_SCHEMA",
+  "druid",
+  "ext",
+  "lookups",
+  "sys",
+  "view"
+]
+```
+
+#### retrieve list of table names in schema
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:411]:
```suggestion
#### Retrieve list of table names in schema
```

File: docs/development/extensions-core/catalog.md [hunk 17]
```diff
@@ -0,0 +1,471 @@
...
+
+retrieve list table names in schema
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:413]:
```suggestion
Retrieve a list of table names in the schema.
```

File: docs/development/extensions-core/catalog.md [hunk 18]
```diff
@@ -0,0 +1,471 @@
...
+
+`GET` `/druid/coordinator/v1/catalog/schemas/{schema}/table
+
+##### Responses
+
+<Tabs>
+
+<TabItem value="1" label="200 SUCCESS">
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:425]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 19]
```diff
@@ -0,0 +1,471 @@
...
+*Successfully retrieved list of table names belonging to schema*
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:428]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 20]
```diff
@@ -0,0 +1,471 @@
...
+</TabItem>
+<TabItem value="2" label="400 BAD REQUEST">
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:432]:
```suggestion
```

File: docs/development/extensions-core/catalog.md [hunk 21]
```diff
@@ -0,0 +1,471 @@
...
+```json
+{
+    "error": "A well-defined error code.",
+    "errorMessage": "A message with additional details about the error."
+}
+```
+</TabItem>
+<TabItem value="3" label="500 INTERNAL SERVER ERROR">
+
+
```
Comment (amaechler) [inline docs/development/extensions-core/catalog.md:444]:
```suggestion
```

**Review by amaechler — COMMENTED** [review]
Only nits, thanks for adding!

**Comment by zachjsh** [comment]
> Only nits, thanks for adding!

Thanks for reviewing @amaechler ! Think I addressed all of your comments. Good suggestions.

**Review by amaechler — APPROVED** [review]

**Pull request merged**